alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/service/ProcessorAttributeService.java

```diff
@@ -0,0 +1,85 @@
+package eu.xenit.alfresco.processor.service;
+
+import lombok.AllArgsConstructor;
+import org.alfresco.service.cmr.attributes.AttributeService;
+import org.alfresco.service.cmr.attributes.DuplicateAttributeException;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.util.ParameterCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+/**
+ * {@link AttributeService} wrapper that allows different Alfresco nodes to claim ranges of transactions.
+ */
+public class ProcessorAttributeService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProcessorAttributeService.class);
+
+    public static final String ATTR_KEY_HEALTH_PROCESSOR = "HealthProcessor";
+    public static final String ATTR_KEY_IS_RUNNING = "IsRunning";
+
+    protected final AttributeService attributeService;
+    protected final DescriptorService descriptorService;
+
+    public <T extends Serializable> T getAttribute(final String key, final T defaultValue) {
+        final List<T> returnValues = new ArrayList<>();
+
+        if(!attributeService.exists(ATTR_KEY_HEALTH_PROCESSOR,key)) {
+            return defaultValue;
+        }
+
+        attributeService.getAttributes((Long id, Serializable value, Serializable[] keys) -> {
+            logger.debug("Processing attribute id: '{}', keys: '{}', value: '{}'", id, keys, value);
+            returnValues.add(extractTransactionStartId(key, keys));
+            return true;
+        }, ATTR_KEY_HEALTH_PROCESSOR, key);
+
+        return returnValues.stream().findFirst().get();
+    }
+
+    public <T extends Serializable> void persistAttribute(final String key, final T value) {
+        ParameterCheck.mandatory("value", value);
+        try {
+            attributeService.createAttribute(getRepositoryId(), ATTR_KEY_HEALTH_PROCESSOR, key, value);
+        } catch (DuplicateAttributeException e) {
+            logger.info("Failed to persist attribute '{}'", value);
+            logger.debug("Cause", e);
+        }
+    }
+
+    public void cleanupAttributes() {
+        attributeService.removeAttributes(ATTR_KEY_HEALTH_PROCESSOR);
+    }
+
+    String getRepositoryId() {
+        return descriptorService.getCurrentRepositoryDescriptor().getId();
+    }
+
+    static <T extends Serializable> T extractTransactionStartId(final String key, Serializable[] attributeKeys){
+        ParameterCheck.mandatory("attributeKeys", attributeKeys);
+        if (attributeKeys.length != 3
```

Comment:
I don't like these integers

Can we deserialize into a POJO and for example use a getter instead of the attributeKeys[0] below?
Or at least statically define these ints in a variable somewhere?

Reply:
I would not start with deserializing these keys into a POJO. 
I agree for static variables with meaning.

Reply:
Added constants that describe the intention of the magic values